packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -114,6 +119,15 @@ export function addTreeToSummary(summary: ISummaryTreeWithStats, key: string, su
     summary.stats = mergeStats(summary.stats, summarizeResult.stats);
 }
 
+export function addHandleToSummary(
```

Comment:
It may be better to add a function `addToSummary` which works for both tree and handle. See `addWithStats` in `SummaryTreeBuilder` below which is what you want but outside of it.